compiler/qsc_passes/src/entry_point.rs

```diff
@@ -139,6 +145,9 @@ impl<'a> Visitor<'a> for EntryPointFinder<'a> {
             if item.attrs.iter().any(|a| a == &Attr::EntryPoint) {
                 self.callables.push((callable, item.id));
             }
+            if callable.name.name.as_ref().to_lowercase() == "main" {
+                self.main.push((callable, item.id));
```

Comment:
Should we validate that the main has no params?

Reply:
That happens automatically above; this part just collects candidates, the other part checks whether they are unique and valid.